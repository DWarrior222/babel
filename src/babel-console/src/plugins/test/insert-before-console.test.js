// import pluginTester from 'babel-plugin-tester';
const pluginTester = require('babel-plugin-tester').default;
const consolePlugin = require('../index');

pluginTester({
  plugin: consolePlugin,
  babelOptions: {
    parserOpts: {
      sourceType: 'unambiguous'
    }
  },
  tests: {
    'console.xx前插入了CallExpression的AST': {
      code: `
        let a = 1;
        console.log(a);
        [1, 2].map(d => d);
        console.error('error');

        {
          const cc = 3;
          console.log(cc, 'cc');
        }
      `,
      snapshot: true,
    }
  }
})