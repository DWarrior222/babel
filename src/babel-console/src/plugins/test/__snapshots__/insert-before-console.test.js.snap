// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unknown plugin console.xx前插入了CallExpression的AST: console.xx前插入了CallExpression的AST 1`] = `

let a = 1;
console.log(a);
[1, 2].map(d => d);
console.error('error');

{
  const cc = 3;
  console.log(cc, 'cc');
}

      ↓ ↓ ↓ ↓ ↓ ↓

let a = 1;
console.log("/Users/guanshuaijie/code/babel/babel-console: (2, 0)");
console.log(a);
[1, 2].map((d) => d);
console.log("/Users/guanshuaijie/code/babel/babel-console: (4, 0)");
console.error("error");
{
  const cc = 3;
  console.log("/Users/guanshuaijie/code/babel/babel-console: (8, 2)");
  console.log(cc, "cc");
}


`;
